api/src/main/java/org/openmrs/module/ModuleFactory.java

```diff
@@ -1169,10 +1169,7 @@ public static void stopModule(Module mod, boolean isShuttingDown) {
 						String extId = ext.getExtensionId();
 						try {
 							List<Extension> tmpExtensions = getExtensions(extId);
-							if (tmpExtensions == null) {
-								tmpExtensions = new Vector<Extension>();
-							}
-							
+							tmpExtensions = new Vector<Extension>();
```

Comment:
Are you sure about this line? Seems like it would change the behavior - `tmpExtensions` would always end up as a new Vector. 

With the old code, it's only initialized to a new Vector if `getExtensions()` fails.